pyuvdata/uvdata/uvh5.py

```diff
@@ -474,30 +472,62 @@ def _read_header(
                 self._set_flex_spw()
         if "flex_spw_id_array" in header:
-            self.flex_spw_id_array = header["flex_spw_id_array"][()]
+            self.flex_spw_id_array = header["flex_spw_id_array"][:]
+        if "multi_phase_center" in header:
+            if bool(header["multi_phase_center"][()]):
+                self._set_phased()
+                self._set_multi_phase_center(preserve_phase_center_info=False)
+        if "Nphase" in header:
+            self.Nphase = int(header["Nphase"][()])
 
-        # check for phasing information
         self.phase_type = bytes(header["phase_type"][()]).decode("utf8")
+        self.object_name = bytes(header["object_name"][()]).decode("utf8")
+        # Here is where we start handing phase center information.  If we have a multi
+        # phase center dataset, this information is going to be handled a bit
+        # differently, since all of these entries are expected to be required arrays.
+        if self.multi_phase_center:
+            self.phase_center_id_array = header["phase_center_id_array"][:]
+
+        # check for phasing information
         if self.phase_type == "phased":
-            self._set_phased()
             self.phase_center_ra = float(header["phase_center_ra"][()])
             self.phase_center_dec = float(header["phase_center_dec"][()])
-            self.phase_center_epoch = float(header["phase_center_epoch"][()])
-            if "phase_center_frame" in header:
-                self.phase_center_frame = bytes(
-                    header["phase_center_frame"][()]
-                ).decode("utf8")
-        elif self.phase_type == "drift":
-            self._set_drift()
+            if "phase_center_epoch" in header:
+                self.phase_center_epoch = float(header["phase_center_epoch"][()])
+            self._set_phased()
+            if "phase_center_app_ra" in header and "phase_center_app_dec" in header:
+                self.phase_center_app_ra = header["phase_center_app_ra"][:]
+                self.phase_center_app_dec = header["phase_center_app_dec"][:]
+            if "phase_center_frame_pa" in header:
+                self.phase_center_frame_pa = header["phase_center_frame_pa"][:]
         else:
-            self._set_unknown_phase_type()
+            if self.phase_type != "drift":
+                warnings.warn(
+                    "Unknown phase types are no longer supported, marking this "
+                    'object as phase_type="drift" by default. If this was a phased '
+                    "object, you can set the phase type correctly by running the "
+                    "_set_phased() method."
+                )
+            self._set_drift()
+
+        # Here is where we collect the other optional source/phasing info
+        if "phase_center_catalog" in header:
+            self.phase_center_catalog = {}
+            for key in header["phase_center_catalog"].keys():
+                self.phase_center_catalog[key] = literal_eval(
```

Comment:
There may be other ways of implementing this, although I think that for a `dict`, this is by far the most clean and straightforward way of importing these. Note that `literal_eval` is not the same as `eval` -- it nominally shouldn't allow one to execute random code, but rather can only hand back objects from a limited range of types (strings, numbers, tuples, lists, dicts, booleans, and None). As such, I don't think it's dangerous to have in here, but we can chat about it further.

Reply:
I think I'm mostly curious what is being `eval`ed here? Does this catalog entry come in a fixed form'? I am also not terribly comfortable with any form of eval, but mostly out of ignorance because anything function with `eval` in the name clearly must not be safe :P

Reply:
Based on our discussion today, I'll be using python's JSON library (specifically the `dumps` and `loads` functions) for string handling instead of `literal_eval` and `str`.

Reply:
Changed in ba624469.